browser/net/brave_network_audit_search_ad_browsertest.cc

```diff
@@ -0,0 +1,129 @@
+  BraveNetworkAuditSearchAdTest& operator=(
+      const BraveNetworkAuditSearchAdTest&) = delete;
+
+  void SetUpOnMainThread() override {
+    InProcessBrowserTest::SetUpOnMainThread();
+    mock_cert_verifier_.mock_cert_verifier()->set_default_result(net::OK);
+    host_resolver()->AddRule("*", "127.0.0.1");
+
+    https_server_ = std::make_unique<net::EmbeddedTestServer>(
+        net::test_server::EmbeddedTestServer::TYPE_HTTPS);
+    https_server_->SetSSLConfig(net::EmbeddedTestServer::CERT_OK);
+
+    base::FilePath test_data_dir =
+        base::PathService::CheckedGet(brave::DIR_TEST_DATA);
+    test_data_dir = test_data_dir.AppendASCII(kEmbeddedTestServerDirectory);
+    https_server_->ServeFilesFromDirectory(test_data_dir);
+
+    ASSERT_TRUE(https_server_->Start());
+  }
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    base::FilePath source_root_path =
+        base::PathService::CheckedGet(base::DIR_SRC_TEST_DATA_ROOT);
+
+    // Full log containing all the network requests.
+    net_log_path_ = source_root_path.AppendASCII("network_search_ad_log.json");
+
+    // Log containing the results of the audit only.
+    audit_results_path_ =
+        source_root_path.AppendASCII("network_audit_search_ad_results.json");
+
+    command_line->AppendSwitchPath(network::switches::kLogNetLog,
+                                   net_log_path_);
+    command_line->AppendSwitchASCII(network::switches::kNetLogCaptureMode,
+                                    "Everything");
+    mock_cert_verifier_.SetUpCommandLine(command_line);
+    InProcessBrowserTest::SetUpCommandLine(command_line);
+  }
+
+  void SetUpInProcessBrowserTestFixture() override {
+    InProcessBrowserTest::SetUpInProcessBrowserTestFixture();
+    mock_cert_verifier_.SetUpInProcessBrowserTestFixture();
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    InProcessBrowserTest::TearDownInProcessBrowserTestFixture();
+    mock_cert_verifier_.TearDownInProcessBrowserTestFixture();
+    std::vector<std::string> allowed_prefixes;
+    for (const char* prefix : kAllowedBraveSearchPrefixes) {
+      allowed_prefixes.push_back(prefix);
+    }
+    VerifyNetworkAuditLog(net_log_path_, audit_results_path_, allowed_prefixes);
+  }
+
+  Profile* profile() { return browser()->profile(); }
+  net::EmbeddedTestServer* https_server() { return https_server_.get(); }
+
+ private:
+  content::ContentMockCertVerifier mock_cert_verifier_;
+  std::unique_ptr<net::EmbeddedTestServer> https_server_;
+
+  base::FilePath net_log_path_;
+  base::FilePath audit_results_path_;
+};
+
+IN_PROC_BROWSER_TEST_F(BraveNetworkAuditSearchAdTest, SearchAdTest) {
+  GURL url = https_server()->GetURL(kDomain, kBraveSearchPath);
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+  content::WebContents* contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+  WaitForLoadStop(contents);
```

Comment:
`NavigateToURL` already makes this call inside, it navigates until loading stops